controlplane/kubeadm/internal/cluster.go

```diff
@@ -97,9 +97,9 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 	}
 	restConfig.Timeout = 30 * time.Second
 
-	c, err := client.New(restConfig, client.Options{Scheme: scheme.Scheme})
+	c, err := m.Tracker.GetClient(ctx, clusterKey)
```

Comment:
This could panic if `Tracker` is nil, should we check before hand?
```
if m.Tracker == nil {
  return nil, errors.New("Cannot get WorkloadCluster: No remote Cluster Cache")
}
```

Reply:
Good idea! Fixed.